### ISSUE TITLE
request: use Spotify full version numbers

### ISSUE THREAD

**Issue by Tafkaroll** [autopkg/homebysix-recipes#340]
Request for not splitting or removing parts of the version numbering in the Spotify recipe.

**Comment by jgstew** [comment]
Are you saying you want to see `1.1.45.621.gdddebadc` instead of `1.1.45.621` as the version?

I guess I don't see why you would want to see `1.1.45.621.gdddebadc` since that is unusual.

**Comment by Tafkaroll** [comment]
Why is it unusual to not use the full version numbering?

**Comment by jgstew** [comment]
because `gdddebadc` cannot be easily compared. `1.1.45.621` should be the full semantic version of the release, while `gdddebadc` seems like a build identifier or commit message identifier or ... something?

Generically you would do a version comparison to say existing spotify installed is < `1.1.45.621` but doing a version comparison with `1.1.45.621.gdddebadc` doesn't work well.

Is there an instance of Spotify in which the only difference in version is on the `.gdddebadc` side of it in this example, but the `1.1.45.621` side is the same?

**Comment by jgstew** [comment]
Recipe change:

```
	<string>com.github.homebysix.pkg.Spotify</string>
	<key>Input</key>
	<dict>
		<key>NAME</key>
		<string>Spotify</string>
		<key>VERSION_SPLIT_ON</key>
		<string>.g</string>
	</dict>
	<key>MinimumVersion</key>
	<string>0.2.0</string>
	<key>ParentRecipe</key>
	<string>com.github.autopkg.download.Spotify</string>
	<key>Process</key>
	<array>
		<dict>
			<key>Arguments</key>
			<dict>
				<key>dmg_path</key>
				<string>%pathname%</string>
			</dict>
			<key>Processor</key>
			<string>AppDmgVersioner</string>
		</dict>
		<dict>
			<key>Arguments</key>
			<dict>
				<key>split_on</key>
				<string>%VERSION_SPLIT_ON%</string>
			</dict>
			<key>Processor</key>
			<string>com.github.homebysix.VersionSplitter/VersionSplitter</string>
		</dict>
```

Then you can override the input key `VERSION_SPLIT_ON` to `DoNotSplitFakeString` and then it does what you want.

**Comment by jgstew** [comment]
This pull request moves the version split on string as an input to the recipe so that you can override it more easily: https://github.com/autopkg/homebysix-recipes/pull/390

**Comment by homebysix** [comment]
Fixed in #390. Default behavior remains as is, but the new `VERSION_SPLIT_ON` input variable allows using the full version if desired.
